### PR TITLE
SOURCE_DIR option disable call to find_package()

### PR DESCRIPTION
**Pull request by ClausKlein**
this fix https://github.com/cpm-cmake/CPM.cmake/issues/455

### CODE REVIEW

File: cmake/CPM.cmake [hunk 1]
```diff
@@ -648,19 +648,21 @@ function(CPMAddPackage)
     return()
   endif()
 
-  if(CPM_USE_LOCAL_PACKAGES OR CPM_LOCAL_PACKAGES_ONLY)
-    cpm_find_package(${CPM_ARGS_NAME} "${CPM_ARGS_VERSION}" ${CPM_ARGS_FIND_PACKAGE_ARGUMENTS})
+  if(NOT DEFINED CPM_ARGS_SOURCE_DIR)
```
Comment (TheLartians) [inline cmake/CPM.cmake:651]:
```suggestion
  if(NOT CPM_ARGS_FORCE)
```

**Review by TheLartians — COMMENTED** [review]
Interesting edge case, tbh I think using the local package should be the expected behaviour when `CPM_USE_LOCAL_PACKAGES` or `CPM_LOCAL_PACKAGES_ONLY` is defined. How about instead instead using the existing `FORCE` argument to disable env based overrides?

Then packages that should **always** be added the defined way can set the additional parameter `FORCE YES`.

File: cmake/CPM.cmake (hunk 1 above)
Comment (ClausKlein) [reply cmake/CPM.cmake:651]: Neither `FORCE` nor `SOURCE_DIR` are documented?

And the `SOURCE_DIR` is use at `ModernCppStarter`!

**Comment by ClausKlein** [comment]
> Interesting edge case, tbh I think using the local package should be the expected behaviour when `CPM_USE_LOCAL_PACKAGES` or `CPM_LOCAL_PACKAGES_ONLY` is defined. How about instead instead using the existing `FORCE` argument to disable env based overrides?
> 
> Then packages that should **always** be added the defined way can set the additional parameter `FORCE YES`.

Yes, may be a clear Solution, but incompatible to the [SOURCE_DIR option use at ModernCppStarter](https://github.com/TheLartians/ModernCppStarter/blob/84ed75ca7db28672a29d865382389ad3e0f03f8a/standalone/CMakeLists.txt#L19)

File: cmake/CPM.cmake (hunk 1 above)
Comment (TheLartians) [reply cmake/CPM.cmake:651]: > Neither `FORCE` nor `SOURCE_DIR` are documented yet?

`SOURCE_DIR` is an origin parameter of [ExternalProject](https://cmake.org/cmake/help/latest/module/ExternalProject.html#id1), so documentation can be found there. `FORCE` we should definitely document though.

**Comment by TheLartians** [comment]
> Yes, may be a clear Solution, but incompatible to the [SOURCE_DIR option used at ModernCppStarter](https://github.com/TheLartians/ModernCppStarter/blob/84ed75ca7db28672a29d865382389ad3e0f03f8a/standalone/CMakeLists.txt#L19)

Yeah it seems that the starter currently doesn't support `CPM_USE_LOCAL_PACKAGES` or `CPM_LOCAL_PACKAGES_ONLY`. I think adding the `FORCE` option there would make sense.

**Comment by ClausKlein** [comment]
Does this changes have a site effect to `CPMFindPackage()`, I think yes!

File: README.md [hunk 2]
```diff
@@ -199,6 +199,9 @@ This can be controlled on a per package basis with the `CPM_DOWNLOAD_<dependency
 ### CPM_USE_LOCAL_PACKAGES
 
 CPM can be configured to use `find_package` to search for locally installed dependencies first by setting the CMake option `CPM_USE_LOCAL_PACKAGES`.
+
+If a packages should always be added the defined way you should set the additional parameter FORCE YES to disable the use of `find_package`
+
```
Comment (TheLartians) [inline README.md:204]: Not sure this is easy to understand. How about instead adding the following at the end of the `CPM_USE_LOCAL_PACKAGES` section (after line 208)?

```markdown
Note that this does not apply to dependencies that have been defined with a truthy `FORCE` parameter. These will be added as defined.
```

**Review by TheLartians — COMMENTED** [review]
Thanks for the update!

File: README.md (hunk 2 above)
Comment (ClausKlein) [reply README.md]: done

**Review by TheLartians — APPROVED** [review]

**Pull request merged**